### PR TITLE
adds masonry for dorfs!

### PR DESCRIPTION
**Pull request by cakexensen**
## What is changing?

adds stone equivalents of some basic furniture and constructions for lavaland dorfs. now dorfs can use more lore-appropriate things when decorating and expanding their fortress. new weird tool for deconstructing these things, instead of giving dorfs a toolbox. stone is useless for station crew and they probably won't encounter it.

masonry

### Changes

- new ore: stone, byproduct of smelting (one for one).
- new sheet: smooth stone, refined from stone ore with hammer.
- new furniture: stone throne, stone table, stone bed, stone coffer.
- new constructions: stone window, stone wall, stone girder, stone floor tiles.
- "dwarf tool" to deconstruct things made of stone (instead of wrench etc).
- can make carving blocks from stone.
- can make chisel from stone.

### Wiki

Stone: After smelting ore, stones will drop from the smelter. Hammer these with the smith's hammer or the strange gold tool to smooth them. Smooth stones can be made into various furniture and constructions. Some of the recipes are in the Dwarf Crafting book. Make your dream fortress with a true Dwarf aesthetic!

Tool: Dwarven constructions are made of stone, and can't be deconstructed with your typical spaceperson's tools. You will find one gelded knapper-gimlet (what an antiquated name!) AKA the Dwarf tool alongside the smith's hammers. Just click on (almost) anything made of stone to deconstruct it. Wow!

## Why these changes?

dorfs didn't feel dorfy enough with all that wood sitting around (although it does anger the filthy elves...), so here's a bunch of stone stuff. now dorfs can actually expand their fortress in style, or repair damage from the occasional megafa--i mean forgotten beast. you have plenty of toys now, stay off the dang station.

### CODE REVIEW

File: russstation/code/game/structures/stone.dm [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+/obj/structure/chair/attackby(obj/item/I, mob/user, params)
+	if(I.tool_behaviour == TOOL_DWARF)
+		to_chat(user, span_notice("You begin deconstructing \the [src]..."))
+		I.play_tool_sound(src)
+		if(I.use_tool(src, user, 1 SECONDS, volume=30))
+			deconstruct()
+			user.visible_message(span_notice("[user] tears apart \the [src]."),
+				span_notice("You successfully disassemble \the [src]."),
+				span_hear("You hear stones dropping to the ground."))
+
```
Comment (MrMelbert) [inline russstation/code/game/structures/stone.dm:128]: - This is overriding the base `obj/structure/chair/attackby` proc
- This doesn't call ever parent, which may cause funky behavior given attackby is a signal proc and is kinda important.

File: russstation/code/game/turfs/closed/mineral_walls.dm [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+	add_fingerprint(user)
+
+	var/turf/T = user.loc //get user's location for delay checks
+
+	//the istype cascade has been spread among various procs for easy overriding
+	// no wall mounting for dorfs right now
+	if(try_clean(I, user, T) || try_decon(I, user, T))
+		return
+
+	return // no parent call
```
Comment (MrMelbert) [inline russstation/code/game/turfs/closed/mineral_walls.dm:38]: I know why this doesn't call parent but the fact that it doesn't seems like a bad idea (attackby is a signal proc)

File: russstation/code/datums/components/dwarf_rune.dm [hunk 3]
```diff
@@ -28,14 +29,23 @@
...
 
 	return checkAttack(user)
 
+/datum/component/dwarf_rune/proc/onItemAttackObj(datum/source, atom/movable/target, mob/living/user)
+	SIGNAL_HANDLER
+
+	// for some reason these aren't always the right types?? why tg why
+	if(istype(user))
+		return checkAttack(user)
+	else if(istype(target, /mob/living))
```
Comment (MrMelbert) [inline russstation/code/datums/components/dwarf_rune.dm:43]:
```suggestion
	else if(isliving(target))
```

Comment (MrMelbert) [reply russstation/code/datums/components/dwarf_rune.dm:43]: also I have no idea what you mean by "aren't always the right types" but it sounds cursed

Comment (cakexensen) [reply russstation/code/datums/components/dwarf_rune.dm:43]: i legit don't get it, but i was getting runtimes like `/obj/whatever doesn't have dropItemToGround()` when hitting things with the dwarf tool, which meant that the obj being attacked was in the `user` param. def cursed

**Pull request merged**